### ISSUE TITLE
Display top of the page after back navigation

### ISSUE THREAD

**Issue by karlcow** [mozilla-mobile/firefox-ios#8270]
## Steps to reproduce
1. Go to https://tv2.dk/
2. scroll a couple of times
3. Tap on an article
4. Tap the back navigation

### Expected behavior
Return at the position where the page was left

### Actual behavior
Return to the top of the page. The position has been forgotten.

### Device & build information
* Device: iPhone XS
* Build version: Firefox Firefox 33 iOS 14.4.1
### Notes
Initial report on https://github.com/webcompat/web-bugs/issues/68763

**Comment by ericswpark** [comment]
Also reproducible on DuckDuckGo. Scroll in the results page, click on a link, scroll all the way down in the linked page, and come back. I expect the scroll position to be remembered, but I end up at the end of the DDG results page.

**Comment by colejohnson66** [comment]
Still an issue on 119.1 with iOS 17.0.3.

**Comment by brendanjones** [comment]
Screen recording for this one: 

https://github.com/mozilla-mobile/firefox-ios/assets/16049594/899afdd9-3bd2-4771-8267-3e292e938f51

I wonder if this is linked to https://github.com/mozilla-mobile/firefox-ios/issues/17583?

**Comment by colejohnson66** [comment]
Still an issue. Had it happen a day or two ago

**Comment by brendanjones** [comment]
Can this please be fixed? It’s absolutely infuriating behaviour.

Then main way it shows up is:
1. Be at the top of any web page
2. Tap a link to a new page
3.  scroll down the page so your viewport is no longer at the top of the page
4. Navigate back to the first page
5.  scroll position will be the same as the previous web page, instead of at the top of the page where you left it.

**Comment by DHowett** [comment]
After a while of being _somewhat bearable,_ it seems like this issue has gotten significantly worse in recent updates.

**Comment by brendanjones** [comment]
Yes, agreed. I’d mostly forgotten about it because it only occurred to a small degree and on a couple of websites, but now it’s back with a vengeance. It’s worst on any website with infinite scroll, like Reddit.

**Comment by janbrasna** [comment]
This specific issue was brought over from webcompat QA, which most likely was a site issue, and no longer reproduces (on both https://tv2.dk and https://nyheder.tv2.dk …).

While your experience may sound similar, it's most likely a separate issue that will each need investigating in isolation, incl. your exact devices, versions, STR etc.

For those running into similar issues on iPadOS previously, there is a workaround shipping since 147.0 from January, so that might have resolved you particular use case:
- https://github.com/mozilla-mobile/firefox-ios/issues/30938

For those who were seeing this regress around v120, this was fixed:
- https://github.com/mozilla-mobile/firefox-ios/issues/16643
(incl. navigating to mid-page or following the other tab's scroll position when returning back)

@ericswpark I cannot reproduce this DDG issue following the steps — if you still do, please file a distinct report, incl. your device/versions and ideally recording the steps.

@colejohnson66 GH doesn't do typical bfcache navigation, but scripts its own page transitions, so that's more likely their JS app state. Feel free to file a separate issue with more details ideally capturing the issue in a screen recording — in case there is something to be improved interacting with their state — but it's mostly their code handling the navigation changes.

@brendanjones For desktop-mode wikipedia there's #18081 that however only seems to reproduce just on iPad (maybe largest iPhone would also trigger their 2-col layout repaint? dunno), for Reddit yes I see some pages reloaded to the top position, but that looks like their code, as I can see the same in Safari: https://github.com/user-attachments/assets/9c8ec754-799c-407c-85c3-083be9125b1c
